src/screens/SendToken/SendTokenAmount.js

```diff
@@ -187,57 +204,84 @@ class SendTokenAmount extends React.Component<Props, State> {
   };
 
   handleChange = (value: Object) => {
-    this.setState({ value });
-    this.updateGasLimit();
+    // first update the amount, then after state is updated check for errors
+    this.setState({ value, gettingFee: true });
+    this.getTxFeeInWei()
+      .then(txFeeInWei => this.setState({ txFeeInWei, gettingFee: false }))
+      .catch(() => null);
+    this.checkFormInputErrors();
+    if (!checkIfSmartWalletAccount(this.props.activeAccount)) {
+      this.updateGasLimit();
+    }
   };
 
-  handleFormSubmit = () => {
+  handleFormSubmit = async () => {
+    const { submitPressed } = this.state;
+    if (submitPressed) return;
     this.formSubmitted = true;
-    const txFeeInWei = this.getTxFeeInWei();
+    this.setState({ submitPressed: true });
+    const txFeeInWei = await this.getTxFeeInWei();
     const value = this._form.getValue();
-    const { navigation } = this.props;
-    const { gasLimit } = this.state;
-    const transactionSpeed = this.getTxSpeed();
-    const gasPrice = txFeeInWei.div(gasLimit).toNumber();
     if (!value) return;
-    const transactionPayload: TokenTransactionPayload = {
+    // $FlowFixMe
+    let transactionPayload: TokenTransactionPayload = {
       to: this.receiver,
       amount: value.amount,
-      gasLimit,
-      gasPrice,
       txFeeInWei,
-      txSpeed: transactionSpeed,
       symbol: this.assetData.token,
       contractAddress: this.assetData.contractAddress,
       decimals: this.assetData.decimals,
     };
-
+    const { navigation, activeAccount } = this.props;
+    if (!checkIfSmartWalletAccount(activeAccount)) {
+      const { gasLimit } = this.state;
+      const transactionSpeed = this.getTxSpeed();
+      const gasPrice = txFeeInWei.div(gasLimit).toNumber();
+      transactionPayload = {
+        ...transactionPayload,
+        gasPrice,
+        gasLimit,
+        txSpeed: transactionSpeed,
+      };
+    }
     Keyboard.dismiss();
+    this.setState({ submitPressed: false }); // reset in case of user navigates back
```

Comment:
it's possible that setState would start after we navigate the user to another screen thus causing an error. I would suggest to use `setState` callback:
```suggestion
this.setState({ submitPressed: false }, () => {
  navigation.navigate(SEND_TOKEN_CONFIRM, {
    transactionPayload,
    source: this.source,
  });
});
```

Reply:
👍